Sources/Tracking/Util/Log.swift

```diff
@@ -1,34 +1,49 @@
 import Foundation
+import os.log
+import OSLog
 
-public protocol Logger {
-    func verbose(_ message: String)
+/// mockable logger + abstract that allows you to log to multiple places if you wish
+public protocol Logger: AutoMockable {
+    /// the noisey log level. Feel free to spam this log level with any
+    /// information about the SDK that would be useful for debugging the SDK.
     func debug(_ message: String)
+    /// Not noisy log messages. Good for general information such as
+    /// when the background queue begins and ends running but use `debug`
+    /// for the status of each background queue task running.
     func info(_ message: String)
-    func warning(_ message: String)
+    /// the SDK is in an unstable state that you want to notify
```

Comment:
decided to cut the warn level?

Reply:
Yes. The original list of log levels (info, warning, verbose, etc) mostly came [Android](https://developer.android.com/reference/android/util/Log) TBH. 

[The Swift Logger class](https://developer.apple.com/documentation/os/oslogtype) does not include a `warning` case so I removed it.